src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsRecoveryObject.java

```diff
@@ -14,4 +28,48 @@ public String getRecoveredAdUnitId(String adUnitId) {
     jsActions.waitForJavaScriptTruthy(String.format("window._sp_ && window._sp_.getElementId && %s", getIdScript));
     return jsActions.execute(getIdScript).toString();
   }
+
+  public void verifyPageFairRecovery() {
+    int recoverableAdsCount = 2;
+    By spansBodyChildrenSelector = By.cssSelector("body>span");
+    Dimension topLeaderboardSize = new Dimension(728, 90);
+    Dimension medrecSize = new Dimension(300, 250);
+    String expectedRecoveredLB = null;
+    String expectedRecoveredMR = null;
+
+    try {
+      expectedRecoveredLB = readFileToString(new File(EXPECTED_TOP_LEADERBOARD_PATH));
+      expectedRecoveredMR = readFileToString(new File(EXPECTED_MEDREC_PATH));
+    } catch (IOException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)